### ISSUE TITLE
Figure out why we're leaking npiperelay.exe instances

### ISSUE THREAD

**Issue by anaisbetts** [anaisbetts/ssh-agent-relay#1]
Something isn't being torn down correctly and we're ending up leaking npiperelay.exe instances, we probably need to see if the associated socat is alive and if it's not, trash everything via `taskkill.exe /F /IM npiperelay.exe`

**Comment by anaisbetts** [comment]
Possibly fixed by b68520a914509

**Comment by kylewlacy** [comment]
I'm not sure if it's the same leak that this issue was originally referring to, but I believe I found a potential source of leaks with my own setup.

Basically, I noticed that my WSL2 environment would eventually just hang on any network requests indefinitely, and looking in Task Manager revealed it was because there were too many instances of `ssh-agent` and `npiperelay`. Running the command `taskkill.exe /F /IM npiperelay.exe` would temporarily fix things.

VS Code has a setting called [`git.autofetch`](https://code.visualstudio.com/Docs/editor/versioncontrol#_remotes), which (I believe) by default calls `git fetch` every 3 minutes when in a workspace with a git repository. I set up a small shell script that used `tasklist.exe` to watch for how many instances of `npiperelay` and `ssh-agent` were running every mintue, and lo and behold, every 3 minutes both numbers went up by one. Then, I turned `git.autofetch` off and suddenly the numbers stopped going up!

So for me, I had a leak caused by `git.autofetch` in VS Code. So for now I've turned this setting off, but it'd be nice if VS Code and `ssh-agent-relay` were to play nice together!
